File: app/src/main/java/com/vo1d/schedulemanager/v2/data/BaseDao.java
package com.vo1d.schedulemanager.v2.data;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Update;

public interface BaseDao<T> {
    @Insert
    long insert(T obj);

    @Update
    void update(T obj);

    @Delete
    void delete(T... objects);
}

File: app/src/main/kotlin/ru/vo1d/timetablemanager/data/BaseDao.kt
package ru.vo1d.timetablemanager.data

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy.IGNORE
import androidx.room.Update

interface BaseDao<PK, T : DatabaseEntity<PK>> {
    @Insert(onConflict = IGNORE)
    suspend fun insert(items: T): PK

    @Insert(onConflict = IGNORE)
    suspend fun insert(vararg items: T)

    @Update
    suspend fun update(item: T)

    @Delete
    suspend fun delete(vararg items: T)
}